parts/windowsProvision.ps1

```diff
@@ -1,11 +1,253 @@
 # Licensed under the MIT License.
 
-mkdir c:/bin
-mkdir c:/tmp
-cd c:/tmp
+$ErrorActionPreference = "Stop"
 
-function InstallOpenSSH()
+$AZUREDATA_DIRECTORY = Join-Path ${env:SystemDrive} "AzureData"
+$AZUREDATA_BIN_DIRECTORY = Join-Path $AZUREDATA_DIRECTORY "bin"
+$PACKAGES_DIRECTORY = Join-Path $env:TEMP "packages"
+$PACKAGES = @{
+    "openssh" = @{
+        "is_vanilla" = $true
+        "url" = "https://github.com/PowerShell/Win32-OpenSSH/releases/download/v7.7.2.0p1-Beta/OpenSSH-Win64.zip"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "OpenSSH-Win64.zip"
+    }
+    "git" = @{
+        "is_vanilla" = $true
+        "url" = "https://github.com/git-for-windows/git/releases/download/v2.19.1.windows.1/Git-2.19.1-64-bit.exe"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "git-2.19.1-64-bit.exe"
+    }
+    "7z" = @{
+        "is_vanilla" = $true
+        "url" = "https://www.7-zip.org/a/7z1805-x64.msi"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "7z1805-x64.msi"
+    }
+    "vs_buildtools" = @{
+        "url" = "https://aka.ms/vs/15/release/vs_buildtools.exe" 
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "vs_buildtools.exe"
+    }
+    "cmake" = @{
+        "url" = "https://cmake.org/files/v3.13/cmake-3.13.0-rc1-win64-x64.msi"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "cmake-3.13.0-rc1-win64-x64.msi"
+    }
+    "ocaml" = @{
+        "url" = "http://www.ocamlpro.com/pub/ocpwin/ocpwin-builds/ocpwin64/20160113/ocpwin64-20160113-4.02.1+ocp1-msvc64.zip"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "ocpwin64.zip"
+    }
+    "sgx_drivers" = @{
+        "url" = "http://download.windowsupdate.com/d/msdownload/update/driver/drvs/2018/01/af564f2c-2bc5-43be-a863-437a5a0008cb_61e7ba0c2e17c87caf4d5d3cdf1f35f6be462b38.cab"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "sgx_base.cab"
+    }
+    "psw" = @{
+        "url" = "http://registrationcenter-download.intel.com/akdlm/irc_nas/13688/Intel%20SGX%20PSW%20for%20Windows%20v2.1.100.46245.exe"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "Intel%20SGX%20PSW%20for%20Windows%20v2.1.100.46245.exe"
+    }
+    "nuget" = @{
+        "url" = "https://dist.nuget.org/win-x86-commandline/v4.1.0/nuget.exe"
+        "local_file" = Join-Path ${PACKAGES_DIRECTORY} "nuget.exe"   
+    }
+}
+
+
+function Start-LocalPackagesDownload {
+    Write-Output "Downloading all the packages to local directory: $PACKAGES_DIRECTORY"
+    New-Directory ${PACKAGES_DIRECTORY}
+    $is_vanilla = "IS_VANILLA_VM"
+    foreach($pkg in $PACKAGES.Keys) {
+        if(!(Test-Path $PACKAGES[$pkg]["is_vanilla"]) -and ($is_vanilla -eq "true")) {
+            Write-Output "Skipping $PACKAGES[$pkg]["local_file"] on Vanilla VM"
+            continue
+        }
+        if(!$PACKAGES[$pkg]["url"]) {
+            if(!(Test-Path $PACKAGES[$pkg]["local_file"])) {
+                Throw "Package $pkg must be manually downloaded to: $($PACKAGES[$pkg]["local_file"])"

```

Comment:
We don't really have this scenario here. So the `if(!$PACKAGES[$pkg]["url"])` can be removed.